### ISSUE TITLE
Getting the ends (first and last items) of an array

### ISSUE THREAD

**Issue by natecavanaugh** [lodash/lodash#2337]
Hey there,
There may already be a way to accomplish this that doesn't require it's own method, but I couldn't come up with a simple way to grab the first and last item together from an array.
For instance, I was chaining operations on an array, but realized that every operation I could come up with required a lot more boilerplate that the `first` and `last` methods.
As an example:
```js
_.chain([1,2,3,4])
.transform((res, n, i, c) => i === 0 || i === c.length - 1 ? res.push(n) : true)
.value()
```
Which is doable, but downright cryptic.

Would it be worth considering for a method like `_.ends` (or something) that pops off the first and last elements of an array?
Or is there already some way to do this by composing other functions together?

I imagine it would behave something like the following uses:

```js
_.ends([1,2,3]); // [1,3]

_.ends([1,2]); // [1,2]

_.ends([1]); // [1]

_.ends([]); // []
```

Thanks for your time :)

**Comment by jdalton** [comment]
Thanks @natecavanaugh!

We'll keep an eye on [the popularity of the request](https://github.com/lodash/lodash/blob/master/.github/CONTRIBUTING.md#feature-requests).

**Comment by StreetStrider** [comment]
I think this issue is too specialized, but it can be solved simply if #2315 will be introduced. This may be treated as :thumbsup: for that issue.

**Comment by jdalton** [comment]
Nice! I was thinking this could be accomplished with some form of composition :)
